core/state/dump.go

```diff
@@ -177,7 +179,13 @@ func (s *StateDB) DumpToCollector(c DumpCollector, conf *DumpConfig) (nextKey []
 		}
 		if !conf.SkipStorage {
 			account.Storage = make(map[common.Hash]string)
-			tr, err := obj.getTrie()
+			var tr Trie
+			if conf.StateScheme == rawdb.PathScheme {
+				tr, err = trie.NewStateTrie(trie.StorageTrieID(obj.db.originalRoot, common.BytesToHash(it.Key),
+					obj.data.Root), obj.db.db.TrieDB())
```

Comment:
The `address` maybe also be nil in `c.OnAccount(address, account)` of the 208 line.

Reply:
I cannot understand, I don't use address here.

Reply:
```
// OnAccount implements DumpCollector interface
func (d *Dump) OnAccount(addr *common.Address, account DumpAccount) {
	if addr != nil {
		d.Accounts[*addr] = account
	}
}
```
if addr is nil, OnAccout may be just a no-op.

Reply:
In bsc, this is no-op, because we don'e use preimages.